pkg/console/subresource/deployment/deployment.go

```diff
@@ -92,110 +88,50 @@ func DefaultDeployment(operatorConfig *operatorv1.Console, cm *corev1.ConfigMap,
 	util.AddOwnerRef(deployment, util.OwnerRefFrom(operatorConfig))
 	return deployment
 }
 
 func DefaultDownloadsDeployment(operatorConfig *operatorv1.Console, infrastructureConfig *configv1.Infrastructure) *appsv1.Deployment {
-	labels := util.LabelsForDownloads()
-	meta := util.SharedMeta()
-	meta.Labels = labels
-	meta.Name = api.OpenShiftConsoleDownloadsDeploymentName
+	downloadsDeployment := resourceread.ReadDeploymentV1OrDie(assets.MustAsset("deployments/console-downloads-deployment.yaml"))
 	replicas := Replicas(infrastructureConfig)
-	affinity := downloadsPodAffinity(infrastructureConfig)
-	gracePeriod := int64(0)
-
-	downloadsDeployment := &appsv1.Deployment{
-		ObjectMeta: meta,
-		Spec: appsv1.DeploymentSpec{
-			Replicas: &replicas,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: labels,
-			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: api.OpenShiftConsoleDownloadsDeploymentName,
-					Annotations: map[string]string{
-						workloadManagementAnnotation: workloadManagementAnnotationValue,
-					},
-					Labels: labels,
-				},
-				Spec: corev1.PodSpec{
-					NodeSelector: map[string]string{
-						"kubernetes.io/os": "linux",
-					},
-					Affinity:                      affinity,
-					Tolerations:                   tolerations(),
-					PriorityClassName:             "system-cluster-critical",
-					TerminationGracePeriodSeconds: &gracePeriod,
-					Containers: []corev1.Container{
-						{
-							Name:                     "download-server",
-							TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
-							Image:                    util.GetImageEnv("DOWNLOADS_IMAGE"),
-							ImagePullPolicy:          corev1.PullPolicy("IfNotPresent"),
-							Ports: []corev1.ContainerPort{{
-								Name:          api.DownloadsPortName,
-								Protocol:      corev1.ProtocolTCP,
-								ContainerPort: api.DownloadsPort,
-							}},
-							ReadinessProbe: downloadsReadinessProbe(),
-							LivenessProbe:  defaultDownloadsProbe(),
-							Command:        []string{"/bin/sh"},
-							Resources: corev1.ResourceRequirements{
-								Requests: map[corev1.ResourceName]resource.Quantity{
-									corev1.ResourceCPU:    resource.MustParse("10m"),
-									corev1.ResourceMemory: resource.MustParse("50Mi"),
-								},
-							},
-							Args: downloadsContainerArgs(),
-						},
-					},
+	downloadsDeployment.Spec.Replicas = &replicas
+	downloadsDeployment.Spec.Template.ObjectMeta.Annotations = map[string]string{
+		workloadManagementAnnotation: workloadManagementAnnotationValue,
+	}
+	downloadsDeployment.Spec.Template.Spec.Affinity = downloadsPodAffinity(infrastructureConfig)
+	downloadsDeployment.Spec.Template.Spec.Containers = []corev1.Container{
+		{
+			Name:                     "download-server",
+			TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+			Image:                    util.GetImageEnv("DOWNLOADS_IMAGE"),
```

Comment:
@jhadvig I can move these 2 containers to the bindata, but would need to sub the image info in. Does that seem reasonable to you?

Reply:
So I would put all the static content in a bindata and the dynamicly generated parts (volumes,...) replace. Check how its done in the [cluster-auth-operator](https://github.com/openshift/cluster-authentication-operator/blob/19971c61b37a47feb2578674f1ae41cc7bee842f/pkg/operator/workload/sync_openshift_oauth_apiserver.go#L134-L152)